nautobot/extras/models/groups.py

```diff
@@ -0,0 +1,283 @@
+    # It's a complex field only modified by the DynamicGroupForm internals at
+    # this time. I am not happy with this pattern right now but due to the
+    # dynamism of the form by merging in the `FooFilterForm` fields, there's not
+    # currently an easy way to move this construction logic to the model.
+    filter = models.JSONField(
+        encoder=DjangoJSONEncoder,
+        editable=False,
+        default=dict,
+        blank=True,
+        null=True,
+        help_text="A JSON-encoded dictionary of filter parameters for group membership",
+    )
+
+    objects = DynamicGroupQuerySet.as_manager()
+
+    clone_fields = ["content_type", "filter"]
+
+    class Meta:
+        ordering = ["content_type", "name"]
+
+    def __str__(self):
+        return self.name
+
+    def get_queryset(self, **kwargs):
+        """
+        Define custom queryset for group model.
+
+        Any `kwargs` are passed along to `DynamicGroupMap.get_queryset()`.
+        """
+
+        model = self.content_type.model_class()
+
+        if not self.filter:
+            return model.objects.none()
+
+        qs = self.map.get_queryset(self.filter, **kwargs)
+
+        # Make sure that this instance can't be a member of its own group
+        if self.present_in_database and model == self.__class__:
+            qs = qs.exclude(pk=self.pk)
+
+        return qs
+
+    @property
+    def members(self):
+        return self.get_queryset()
+
+    @property
+    def count(self):
+        """Return the number of objects in the group."""
+        return self.get_queryset().count()
+
+    @property
+    def map(self):
+        """
+        Accessor to automatically generated `BaseDynamicGroupMap` class.
+
+        This class object is cached on the instance after the first time it is accessed.
+        """
+
+        if getattr(self, "_map", None) is None:
+            try:
+                model = self.content_type.model_class()
+                dynamicgroupmap_class = dynamicgroup_map_factory(model)
+            except models.ObjectDoesNotExist:
+                dynamicgroupmap_class = None
+
+            self._map = dynamicgroupmap_class
+
+        return self._map
+
+    def get_absolute_url(self):
+        return reverse("extras:dynamicgroup", kwargs={"slug": self.slug})
+
+    def get_group_members_url(self):
+        """Get URL to group members."""
+        base_url = self.map.base_url
+        filter_str = self.map.urlencode(self.filter)
+
+        if filter_str is not None:
+            base_url += f"?{filter_str}"
+
+        return base_url
+
+    def get_filter_fields(self):
+        # Do not present the list of filter options until the object has been created and has a map
+        # class.
+        if not self.map:
+            return {}
+
+        if not self.present_in_database:
+            return {}
+
+        # Add all fields defined in the DynamicGroupMap to the form and populate the default value
```

Comment:
"to the form"? What form?

Reply:
This `initial` functionality was replaced with `get_intiial()` so I'm ripping this extra logic out.